gen/build_deploy/powershell/dcos_install.ps1

```diff
@@ -0,0 +1,304 @@
+    if (-not (test-path "$($dir)\RunOnce.ps1") ) {
+        Write-Log("$($basedir)\RunOnce.ps1 missing. Creating")
+        $RunOnceScript_content = "& pip install virtualenv;`n& virtualenv .venv;`n& .venv\Scripts\activate;`n& pip install -r C:\dcos\chocolatey_offline\winpanda\requirements.txt;`n& python C:\dcos\chocolatey_offline\winpanda\cli.py setup --master-private-ipaddr $($masters_ip) --local-private-ipaddr $($winagent_ip)`n"
+        CreateWriteFile "$($basedir)" "RunOnce.ps1" $RunOnceScript_content
+        Write-Log("Creating RunOnce registry record")
+        $KeyName = 'Run'
+        $Command = "%systemroot%\System32\WindowsPowerShell\v1.0\powershell.exe -executionpolicy bypass -file `"$($dir)\RunOnce.ps1`""
+        if (-not ((Get-Item -Path HKLM:\Software\Microsoft\Windows\CurrentVersion\RunOnce).$KeyName )) {
+            New-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\RunOnce' -Name $KeyName -Value $Command -PropertyType ExpandString
+        }
+        else {
+            Set-ItemProperty -Path 'HKLM:\Software\Microsoft\Windows\CurrentVersion\RunOnce' -Name $KeyName -Value $Command -PropertyType ExpandString
+        }
+    } else {
+        Write-Log("$($dir)\RunOnce.ps1 already exists. Skipping")
+    }
+}
+
+function CreateRunOnceScheduledTask($dir, $masters_ip, $winagent_ip) {
+    $destination = "$($dir)\RunOnce.ps1"
+    if (-not (Test-Path $destination) ) {
+        Write-Log("$($destination) missing. Creating")
+        $RunOnceScript_content = "& pip install virtualenv;`n& virtualenv .venv;`n& .venv\Scripts\activate;`n& pip install -r C:\dcos\chocolatey_offline\winpanda\requirements.txt;`n& python C:\dcos\chocolatey_offline\winpanda\cli.py setup --master-private-ipaddr $($masters_ip) --local-private-ipaddr $($winagent_ip)`n#Remove a Scheduled task RunOnce, created while initial provision`nif(Get-ScheduledTask -TaskName `"RunOnce`" -TaskPath '\CustomTasks\' -ErrorAction Ignore) { Unregister-ScheduledTask -TaskName `"RunOnce`" -Confirm:`$False }`n"
+        CreateWriteFile "$($dir)" "RunOnce.ps1" $RunOnceScript_content
+
+        Write-Log("Creating Scheduled Task to run Winpanda")
+        $action = New-ScheduledTaskAction -Execute '%systemroot%\System32\WindowsPowerShell\v1.0\powershell.exe' -Argument $destination
+        $trigger = New-ScheduledTaskTrigger -AtStartup
+        $principal = New-ScheduledTaskPrincipal -UserID "NT AUTHORITY\SYSTEM" -LogonType ServiceAccount -RunLevel Highest
+        Register-ScheduledTask -TaskName 'RunOnce' -TaskPath '\CustomTasks\' -Action $action -Trigger $trigger -Principal $principal -Description 'Winpanda task to run once after winagent provision.' -ErrorAction Stop
+    } else {
+        Write-Log("$($dir)\RunOnce.ps1 already exists. Skipping")
+    }
+}
+
+function SetupMesosDNS($masters_ip) {
+    Write-Log("Mesos-DNS setup with following $masters_ip")
+    #TO DO: Mesos-DNS logic to be placed here
+}
```

Comment:
what is the purpose of this method? I understand that this is needed on the long run, but what is the purpose to have it in the first iteration?

Reply:
correct, for now this is down by Ansible in Alpha and in Beta Sprint 1, but it needs to be done in dcos_install.ps1 to simplify offline setup/pre-setup